configs/components/rubygem-nokogiri.rb

```diff
@@ -5,14 +5,18 @@
     # including its dependencies (mini_portile2), installing it on the target, and
     # installing the nokogiri gem using our gem command. The added files from
     # :gem_home were then tar'ed up and packed into this container tarball.
-    pkg.url "http://buildsources.delivery.puppetlabs.net/nokogiri-precompiled-huaweios-#{pkg.get_version}.tar.gz"
-    pkg.md5sum "8a783bdda33cf5f10afe84cf6fd8f28c"
+    pkg.url "http://buildsources.delivery.puppetlabs.net/nokogiri-precompiled-huaweios-#{pkg.get_version}-for-ruby-#{settings[:ruby_version]}.tar.gz"
+    if settings[:ruby_version] == "2.1.9"
```

Comment:
@MosesMendoza while this component does seem a little dirty, I disagree that the three components would be a cleaner solution. The way that I updated this functionality is consistent with all the other components of puppet-agent when checking for platform when platform specific changes are required.

The argument against using different component files would be that if we needed to update something across the board (i.e. something about the install command needed to change) we would need to update it in 3 different places and not one.

I would however like to clean up this component to fit with the https://github.com/puppetlabs/vanagon/blob/master/docs/STYLEGUIDE.md set forth in vanagon. But I think that's out of scope for this PR.